src/include/daos/lru.h

```diff
@@ -158,4 +158,13 @@ daos_lru_ref_add(struct daos_llink *llink)
 	llink->ll_ref++;
 }
 
+/**
+ * Return true if the caller is the last user of the LRU element.
+ */
+static inline bool
+daos_lru_is_last_user(struct daos_llink *llink)
+{
+	return llink->ll_ref <= 2;
```

Comment:
This relies on that hash table holds one refcount, and looks current daos lru implementation assumes the hash table is always created without D_HASH_FT_EPHEMERAL, but I didn't see where how it's guranteed.

If the patch needs be refreshed, please add a D_ASSERT(!(feats & D_HASH_FT_EPHEMERAL)) in daos_lru_cache_create().